homeassistant/components/shelly/__init__.py

```diff
@@ -39,10 +39,22 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
         entry.data.get(CONF_PASSWORD),
         temperature_unit,
     )
+
+    if COAP_CONTEXT not in hass.data[DOMAIN]:
+        coap_context = hass.data[DOMAIN][
+            COAP_CONTEXT
+        ] = await aiocoap.Context.create_client_context()
+        hass.data[DOMAIN][UNDO_SHUTDOWN_LISTENER] = hass.bus.async_listen(
+            EVENT_HOMEASSISTANT_STOP, shutdown_listener(hass)
```

Comment:
dont put a coroutine as a listener

Reply:
Do you mean that `shutdown_listener` should be sync function and run coap context shutdown via `asyncio.run_coroutine_threadsafe`?